### PR TITLE
Refactor the parsing logic

### PR DESCRIPTION
**Pull request by juchi**

### CODE REVIEW

File: src/Html2Pdf.php [hunk 1]
```diff
@@ -64,6 +64,38 @@ class Html2Pdf
...
+     */
+    private $currentNode;
+
+    /**
+     * The Node stack from the root object to the current Node
+     *
+     * @var array
+     */
+    private $parentNodes = array();
+    /**
```
Comment (clement-michelet) [inline src/Html2Pdf.php]: Space before

File: src/Html2Pdf.php [hunk 2]
```diff
@@ -64,6 +64,38 @@ class Html2Pdf
...
+     * Index of the node being processed among its siblings
+     *
+     * @var int
+     */
+    private $childIndex = 0;
+    /**
```
Comment (clement-michelet) [inline src/Html2Pdf.php]: Space before

File: src/Html2Pdf.php [hunk 3]
```diff
@@ -64,6 +64,38 @@ class Html2Pdf
...
+     * Stack of the current child indexes for each node
+     *
+     * @var array
+     */
+    private $childIndexes = array();
+    /**
```
Comment (clement-michelet) [inline src/Html2Pdf.php]: Space before

**Comment by spipu** [comment]
i have test, there is some notice errors, and the renderer is not the same as in the master branch :(

**Comment by spipu** [comment]
do we delete this branch and PR ?

**Pull request closed without merging**